ci/mage/util/docker.go

```diff
@@ -0,0 +1,24 @@
+package util
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+)
+
+func HostDockerDir() (string, error) {
+	if runtime.GOOS != "linux" {
+		// doesn't work on darwin, untested on windows
+		return "", fmt.Errorf("cannot get docker dir on %s", runtime.GOOS)
+	}
```

Comment:
I've made it work in macOS for a long time when this was first added, I don't remember if it was the new multilanguage engine or modules. One thing I had to do though was disable the osxkeychain credential helper (`credsStore`) and put my token directly in the `~/.docker/config.json` file (under `auths`) so it get's accepted by both the macOS host and dagger on the linux vm.

So in this case, mounting my `~/.docker` dir may work, but not with this check here.